Clinical-Genomics/scout
# Fix updating HPO tems

Fix #3430 

The last lines of the hpo.obo file now look like this:

```
...

[Term]
id: HP:5000048
name: Anti-Kv1.4 antibody
def: "The presence of autoantibodies (immunoglobulins) in the blood circulation that react against the voltage-gated potassium channel Kv1.4." [PMID:22167224, PMID:30918333]
is_a: HP:0030057 ! Autoimmune antibody positivity
created_by: ORCID:0000-0002-3387-1836

[Typedef]
id: inheres_in
name: inheres in
xref: RO:0000052

[Typedef]
id: inheres_in_part_of
name: inheres in part of
xref: RO:0002314
holds_over_chain: inheres_in part_of

[Typedef]
id: part_of
name: part of
xref: BFO:0000050
```

And Scout tries to parse `[Typedef]` items as if they were genes. A solution is stopping line parsing when one reaches the first [Typedef].



<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>


**How to test**:
1. On main branch, locally, run: `scout --demo update hpo` . It will crash
2. Repeat from this branch and HPO terms should be updated

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
